Remove ld_libs_general from ldflags to fix link order with --as-needed.
Libraries must come after object files on the linker command line.
PLD-specific.

diff -Nru xemacs-21.1.8/lib-src/Makefile.in.in xemacs-21.1.8.new/lib-src/Makefile.in.in
--- xemacs-21.1.8/lib-src/Makefile.in.in	Mon Jun 14 08:32:00 1999
+++ xemacs-21.1.8.new/lib-src/Makefile.in.in	Thu Dec  9 05:20:16 1999
@@ -140,7 +140,7 @@
 ## when #include'ing generated files.
 cppflags = -I. -I../src -I$(srcdir) -I$(top_srcdir)/src $(CPPFLAGS)
 cflags   = $(CFLAGS) $(cppflags) $(c_switch_general)
-ldflags  = $(LDFLAGS) $(ld_switch_general) $(ld_libs_general)
+ldflags  = $(LDFLAGS) $(ld_switch_general)
 
 ## This is the default compilation command.
 ## But we should never rely on it, because some make version
